src/main/java/org/jabref/logic/git/GitHandler.java

```diff
@@ -171,34 +171,80 @@ public void mergeBranches(String targetBranch, String sourceBranch, MergeStrateg
         this.checkoutBranch(currentBranch);
     }
 
+     public void pushCommitsToRemoteRepository() throws IOException, GitAPIException {
+        pushCommitsToRemoteRepository(null);
+     }
+
     /**
      * Pushes all commits made to the branch that is tracked by the currently checked out branch.
      * If pushing to remote fails, it fails silently.
      */
-    public void pushCommitsToRemoteRepository() throws IOException {
-        try (Git git = Git.open(this.repositoryPathAsFile)) {
-            try {
+    public void pushCommitsToRemoteRepository(DialogService dialogService) throws IOException, GitAPIException {
+        try {
+            Git git = Git.open(this.repositoryPathAsFile);
+            String remoteURL = git.getRepository().getConfig().getString("remote", "origin", "url");
+            Boolean isSshRemoteRepository = remoteURL != null ? remoteURL.contains(".git") : false;
+            
+            git.verifySignature();
+
+            if (isSshRemoteRepository) {
+                TransportConfigCallback transportConfigCallback = new SshTransportConfigCallback();
                 git.push()
-                   .setCredentialsProvider(credentialsProvider)
-                   .call();
-            } catch (GitAPIException e) {
+               .setTransportConfigCallback(transportConfigCallback)
+               .call();
+            } else {
+                String gitUsername = "";
+                String gitPassword = "";
+
+                if (dialogService != null) {
+                    gitUsername = dialogService.showInputDialogAndWait(Localization.lang("Git credentials"), Localization.lang("git username")).get();
```

Comment:
Please check how JabRef handles proxy credentials. Username/password stored in preferences. You should do that, too,